pdns/ws-auth.cc

```diff
@@ -1160,16 +1160,23 @@ static void apiServerSearchData(HttpRequest* req, HttpResponse* resp) {
   resp->setBody(doc);
 }
 
-void apiServerFlushCache(HttpRequest* req, HttpResponse* resp) {
+void apiServerCacheFlush(HttpRequest* req, HttpResponse* resp) {
   if(req->method != "PUT")
     throw HttpMethodNotAllowedException();
 
+  string name = req->getvars["domain"];
+  if (name.empty() || name[name.size()-1] != '.') {
```

Comment:
I think I'll go with making this the same generic 'unable to parse' error actually.